vllm/model_executor/layers/fused_moe/modular_kernel.py

```diff
@@ -822,7 +851,8 @@ def forward(
              )
         else:
             # Overlap shared expert compute with all2all dispatch.
-            receiver = self.prepare_finalize.prepare_async(
+            dbo_maybe_run_recv_hook()
+            hook, receiver = self.prepare_finalize.prepare_async(
```

Comment:
Would this work with other prepare_finalize impls? Wouldn't you have to update them? As well as the interface?

Reply:
> Support for pplx and DeepEP HT will be added in follow on PRs. We have them working but this PR is already huge.

Yeah I didn't mean adding support, it's just that you're changing the interface. But if it works I trust :)